lib/publiccloud/ssh_interactive.pm

```diff
@@ -1,67 +1,98 @@
 # SUSE's openQA tests
 #
-# Copyright 2019 SUSE LLC
+# Copyright 2019-2022 SUSE LLC
 # SPDX-License-Identifier: FSFAP
 
 # Summary: Class with helpers related to SSH Interactive mode
 #
-# Maintainer: Pavel Dostal <pdostal@suse.cz>
+# Maintainer: qa-c@suse.de
 
 package publiccloud::ssh_interactive;
+use testapi qw(is_serial_terminal);
 use base Exporter;
 use testapi;
 use Utils::Backends qw(set_sshserial_dev unset_sshserial_dev);
+use version_utils qw(is_tunneled);
 use strict;
 use warnings;
 
 our @EXPORT = qw(ssh_interactive_tunnel ssh_interactive_leave select_host_console);
 
+# Helper call to activate a console and establish the ssh connection therein
+sub establish_tunnel_console {
+    my ($console) = @_;
+
+    select_console("$console");
+    # Note: Don't use script_run here! The serial terminal is set to /dev/sshserial, so every script_run will time out
+    type_string("\n~.\n", max_interval => 1);    # ensure no previous ssh connection is present
+    enter_cmd("clear");
+    enter_cmd('ssh -t sut');
+    # give the ssh connection some time to settle
+    sleep 5;
+}
+
 sub ssh_interactive_tunnel {
     # Establish the ssh interarctive tunnel to the publiccloud instance.
     # Optional arguments: 'force => 1' - reestablish tunnel, also if already established.
 
     my $instance = shift;
-    my %args = testapi::compat_args({cmd => undef}, ['cmd'], @_);
+    my %args = testapi::compat_args({force => 0, reconnect => 0}, ['force', 'reconnect'], @_);
     $args{force} //= 0;
+    $args{reconnect} //= 0;
```

Comment:
i dont think you need those two lines. `testapi::compat_args` should return the ```$VAR1 = 'force';
$VAR2 = 0;
$VAR3 = 'reconnect';
$VAR4 = 1;```

Also the perlpod looks need update

Reply:
ok